src/app/wallets/update-pending-invoices.ts

```diff
@@ -139,21 +140,39 @@ const updatePendingInvoice = async ({
       return true
     }
 
-    const invoicePaid = await walletInvoicesRepo.markAsPaid(paymentHash)
-    if (invoicePaid instanceof Error) return invoicePaid
-
     const displayCurrencyPerSat = await getCurrentPrice()
     if (displayCurrencyPerSat instanceof Error) return displayCurrencyPerSat
 
+    // TODO: this should be a in a mongodb transaction session with the ledger transaction below
```

Comment:
Currently we are treating the ledger as a discrete 'sub domain'. We should not have transactions across sub domain boundaries. If we need transactional consistency guarantees (ie atomic updates) across these entities then we have a design flaw.

My feeling is we should execute the `markAsPaid` call _after_ the ledger update (as ledger is the source of truth). And I don't see an issue with it having eventual consistency.

Not a TODO for this PR - just reflecting on the comment